File: solo-leveling-player-service/src/main/java/com/sleepkqq/sololeveling/player/service/config/customizer/SerializationCustomizer.java
package com.sleepkqq.sololeveling.player.service.config.customizer;

import static com.sleepkqq.sololeveling.player.service.mapper.EnumOrdinalMapper.createMapper;
import static com.sleepkqq.sololeveling.player.service.model.task.TaskProps.TOPICS;
import static com.sleepkqq.sololeveling.player.service.model.user.UserProps.ROLES;

import com.sleepkqq.sololeveling.player.service.model.task.enums.TaskTopic;
import com.sleepkqq.sololeveling.player.service.model.user.enums.UserRole;
import org.babyfish.jimmer.sql.JSqlClient.Builder;
import org.babyfish.jimmer.sql.runtime.Customizer;
import org.springframework.stereotype.Component;

@Component
public class SerializationCustomizer implements Customizer {

  @Override
  public void customize(Builder builder) {
    builder
        .setSerializedPropObjectMapper(TOPICS, createMapper(TaskTopic.class))
        .setSerializedPropObjectMapper(ROLES, createMapper(UserRole.class));
  }
}

File: solo-leveling-player-service/src/main/kotlin/com/sleepkqq/sololeveling/player/service/config/customizer/SerializationCustomizer.kt
package com.sleepkqq.sololeveling.player.service.config.customizer

import com.sleepkqq.sololeveling.player.model.entity.task.TaskProps
import com.sleepkqq.sololeveling.player.model.entity.task.enums.TaskTopic
import com.sleepkqq.sololeveling.player.model.entity.user.UserProps
import com.sleepkqq.sololeveling.player.model.entity.user.enums.UserRole
import com.sleepkqq.sololeveling.player.service.mapper.EnumOrdinalMapper
import org.babyfish.jimmer.sql.JSqlClient.Builder
import org.babyfish.jimmer.sql.runtime.Customizer
import org.springframework.stereotype.Component

@Suppress("unused")
@Component
class SerializationCustomizer : Customizer {

	override fun customize(builder: Builder) {
		builder.apply {
			setSerializedPropObjectMapper(
				TaskProps.TOPICS,
				EnumOrdinalMapper.createMapper(TaskTopic::class.java)
			)
			setSerializedPropObjectMapper(
				UserProps.ROLES,
				EnumOrdinalMapper.createMapper(UserRole::class.java)
			)
		}
	}
}
